microprofile/reactive-messaging-smallrye/config/src/main/java/org/wildfly/microprofile/reactive/messaging/config/ReactiveMessagingConfigSource.java

```diff
@@ -65,6 +76,14 @@ public int getOrdinal() {
 
     @Override
     public Set<String> getPropertyNames() {
-        return PROPERTIES.keySet();
+        return properties.keySet();
+    }
+
+    public static void addProperties(Config config, Map<String, String> properties) {
+        for (ConfigSource source : config.getConfigSources()) {
+            if (source.getName().equals(NAME) && source instanceof ReactiveMessagingConfigSource) {
+                ((ReactiveMessagingConfigSource)source).properties.putAll(properties);
+            }
+        }
```

Comment:
This method could use some comments related to scoping so it's understandable to non-experts.

AIUI the Config object passed in is deployment-scoped, and thus the ReactiveMessagingConfigSource is as well. So this method is populating a deployment-scoped object with deployment-scoped properties from the DUP.

Reply:
Added

Reply:
Thanks.